modules/issue_tracker/php/NDB_Menu_Filter_Form_issue_tracker.class.inc

```diff
@@ -0,0 +1,384 @@
+ */
+class NDB_Menu_Filter_Form_Issue_Tracker extends NDB_Menu_Filter_Form
+{
+    //aka building-table-using-react
+    var $AjaxModule = true;
+
+    /**
+     * Set up the variables required by NDB_Menu_Filter class for constructing
+     * a query
+     *
+     * @return null
+     */
+    function _setupVariables()
+    {
+
+        $user =& User::singleton();
+        $db = Database::singleton();
+
+        $this->query = " FROM issues as i " .
+            "LEFT JOIN candidate c ON (i.candID=c.CandID) " .
+            "LEFT JOIN psc p ON (i.centerID=p.CenterID) " .
+            "LEFT JOIN LorisMenu l on (i.module=l.ID) " .
+            "LEFT JOIN session s ON (i.sessionID = s.ID) " .
+            "LEFT JOIN issues_watching w ON (i.issueID = w.issueID) " .
+            "WHERE i.status != 'closed'";
+
+        if (!$user->hasPermission('access_all_profiles')) {
+            $this->query .= " AND i.centerID=" . $db->quote($user->getCenterID());
+        }
+
+        //note that this needs to be in the same order as the headers array
+        $this->columns = array(
+            'i.issueID as Issue_ID',
+            'i.title as Title',
+            'l.Label as Module',
+            'i.category as Category',
+            'i.reporter as Reporter',
+            'i.assignee as Assignee',
+            'i.status as Status',
+            'i.priority as Priority',
+            'p.Name as Site',
+            'c.PSCID as PSCID',
+            's.Visit_label as Visit_Label',
+            'i.lastUpdate as Last_Update',
+        );
+
+        $this->order_by = 'i.issueID DESC';
+        $this->group_by = 'i.issueID';
+
+        //NDB filter will remove underscores
+        $this->headers = array(
+            'Issue_ID',
+            'Title',
+            'Module',
+            'Category',
+            'Reporter',
+            'Assignee',
+            'Status',
+            'Priority',
+            'Site',
+            'PSCID',
+            'Visit_Label',
+            'Last_Update',
+        );
+
+        $this->validFilters = array(
+            'i.issueID',
+            'l.Label',
+            'i.category',
+            'c.PSCID',
+            's.Visit_label',
+            'i.reporter',
+            'i.assignee',
+            'i.centerID',
+            'w.userID',
+            'i.projectID',
+            'i.status',
+            'i.priority',
+            'keyword',
+            'minDate',
+            'maxDate',
+        );
+
+        $this->formToFilter = array(
+            'issueID' => 'i.issueID',
+            'category' => 'i.category',
+            'module' => 'l.Label',
+            'PSCID' => 'c.PSCID',
+            'visitLabel' => 's.Visit_label',
+            'reporter' => 'i.reporter',
+            'assignee' => 'i.assignee',
+            'site' => 'i.centerID',
+            'projectID' => 'i.projectID',
+            'status' => 'i.status',
+            'priority' => 'i.priority',
+            'keyword' => 'i.comment',
```

Comment:
The search by keyword always throws an error, because `i.comment` is not a field in the `issues` table. I guess you'll have to JOIN `issues_comments` table in order to get the comment field. For exact error message, see below:

```
Unknown column 'i.comment' in 'where clause'

Query: <pre>SELECT COUNT(*) FROM (SELECT i.issueID as Issue_ID, i.title as Title, l.Label as Module, i.category as Category, i.reporter as Reporter, i.assignee as Assignee, i.status as Status, i.priority as Priority, p.Name as Site, c.PSCID as PSCID, s.Visit_label as Visit_Label, i.lastUpdate as Last_Update FROM issues as i LEFT JOIN candidate c ON (i.candID=c.CandID) LEFT JOIN psc p ON (i.centerID=p.CenterID) LEFT JOIN LorisMenu l on (i.module=l.ID) LEFT JOIN session s ON (i.sessionID = s.ID) LEFT JOIN issues_watching w ON (i.issueID = w.issueID) WHERE i.status != 'closed' AND i.issueID = :v_iissueid AND ( i.comment LIKE CONCAT('%', :v_searchkey, '%')) GROUP BY i.issueID  ORDER BY i.issueID DESC) as tmptable LIMIT 1</pre>

```